src/cli/node/json_rpc_client.rs

```diff
@@ -174,16 +189,52 @@ pub fn get_utxo_info(
         }
         table.printstd();
         println!("-----------------------");
+    } else {
+        for utxo_metadata in utxo_info
+            .utxos
+            .into_iter()
+            .sorted_by_key(|um| (um.value, um.output_pointer.clone()))
+        {
+            utxo_sum += utxo_metadata.value;
+            if utxo_metadata.ready_for_collateral {
+                utxo_ready_sum += utxo_metadata.value;
+            } else if utxo_metadata.timelock == 0 {
+                utxo_too_small_sum += utxo_metadata.value;
+            } else {
+                utxo_not_ready_sum += utxo_metadata.value;
+            }
+        }
```

Comment:
If you are going to use the for in the two parts of the if condition clause, I would include the table in the "if long clause" to not repeating the same code lines

Reply:
I incorrectly assumed timelock would be set for blocks that were recently minted. In order to solve this I would have to compare utxo_metadata.value with collateral_min, but I don't know how to access the consensus constants in json_rpc_client.rs.

Are you asking to put the for loop outside of the if/else statement?

Reply:
Timelock is used to create transactions that can not be spent until the specified timestamp, so collateral age is only for being used as a collateral in a data request, not for a normal transaction

I was thinking that in the past, some UtxoInfo fields as "bigger_than_min_counter" were used in other parts of the code, so that counters were useful to do not have to calculate everytime, buuuuut now it is only used for the CLI methods. So if we are going to iterate the array of UtxoMetadata's with and without the --long flag, those counters are useless, so maybe we can remove it and include more info in UtxoMetadata. Currently, in UtxoMetadata contains a field called "ready_for_collateral" which has 3 conditions (bigger than min, collateral mature and a valid timelock), so we can also remove it, and include other fields.

My proposal: 
- Remove counters from UtxoInfo struct and include the collateral_min, that could be removed in a future when we could get the ConsensusConstants from JSON-RPC
- Remove "ready_for_collateral" from UtxoMetadata struct and include "mature_utxo" field that only check the collateral age (aka "all_utxos.included_in_block_number(o).unwrap() <= block_number_limit" condition in create_utxo_metadata function)
- Calculate the readyness of the utxo in the part of the JSON-RPC client checking the 3 conditions:
    - Value > collateral_min
    - Maturity
    - Timelock == 0

What do you think?

According to the other question... yes :P

Reply:
That sounds pretty much like what I wanted to do (the only viable solution I saw at the moment), but I was not sure if you would support adding more data to the UtxoMetadata struct.

As for the collateral_min, I would rather add a boolean stating if it is bigger than the collateral_min, rather than the consensus constant itself (saves some space). But yes, both of these will become obsolete once we have access to getConsensusConstants (and I'll update that when this is the case).

I'll update the pull request ASAP.

PS: I actually tried integrating the getConsensusConstants RPC from @tmpolaczyk but am failing miserably (getting a "Method not found" error). I am missing something when implementing it, which I need to further look into.

Reply:
Dont worry!! As soon as you can. And in a week more or less, @tmpolaczyk will come back from holiday, so let him fight with his PR :P